### PR TITLE
Console refactor

### PR DESCRIPTION
**Pull request by m-yukio**
Refactoring of existing code.
1. Common
   - And the private functions to the methods of Utility class, made available from external code to add the command.
   - The option of command reconstituted in sub-command was set to be held in the nest.
   - From member variables "help" of Command structure, it was to be able to output the help of sub-command.
   - It was to organize the command options handling.
2. command Allocator
   - supported option "-h" and option "help".
3. command Config
   - supported option "-h" and option "help".
4. command DebugMsg
   - supported option "-h" and option "help".
   - Set the help wording options.
5. command Director
   - deleted option "help" code.
   - Set the help wording options.
6. command Exit
   - supported option "-h" and option "help".
7. command FileUtils
   - supported option "-h" and option "help".
   - Quit the error wording output when an invalid option is specified.
   - Set the help wording options.
   - Non-flush option (including the empty) was so as to print when you are specified.
8. command Fps
   - supported option "-h" and option "help".
   - Set the help wording options.
   - Non-on/off option (including the empty) was so as to print when you are specified.
9. command Help
   - supported option "-h" and option "help".
10. command Projection
    - supported option "-h" and option "help".
    - Set the help wording options.
    - Quit the error wording output when an invalid option is specified.
    - Non-2d/3d option (including the empty) was so as to print when you are specified.
11. command Resolution
    - supported option "-h" and option "help".
12. command SceneGraph
    - supported option "-h" and option "help".
13. command Texture
    - supported option "-h" and option "help".
    - Set the help wording options.
    - Quit the error wording output when an invalid option is specified.
    - Non-flush option (including the empty) was so as to print when you are specified.
14. command Touch
    - deleted option "help" code.
    - Set the help wording options.
15. command Upload
    - No change
16. command Version
    - supported option "-h" and option "help".

### CODE REVIEW

**Comment by m-yukio** [comment]
I believe that to facilitate the command added to the console.
If you have this PR are merged, I am scheduled to PR to extend the console commands.

**Comment by minggo** [comment]
@m-yukio Sorry, i don't quite understand this PR. It added some functions declaration, but haven't added implementations.

**Comment by m-yukio** [comment]
@minggo I believe the additional command for the product.
In the existing implementation, because the parameter you want to change is hidden inside, it will become necessary cocos2d-x-side code modification. In addition, the features available in common, we have published as a function to be available from the outside.

**Comment by minggo** [comment]
What i mean is that i don't find any implementation. This PR is not completion?

**Comment by m-yukio** [comment]
@minggo There is a implementation part.
Maybe, because there are a lot of CCConsole.cpp changes, for you will do has been pointed out that the "Files changed" not in appear?
I think that if it is possible to display the view button.

**Comment by minggo** [comment]
@m-yukio yep, sorry about it

File: cocos/base/CCConsole.h [hunk 1]
```diff
@@ -107,28 +180,63 @@ class CC_DLL Console
...
-    void commandDirector(int fd, const std::string &args);
-    void commandTouch(int fd, const std::string &args);
+    void commandAllocator(int fd, const std::string& args);
+    void commandConfig(int fd, const std::string& args);
+    void commandDebugMsg(int fd, const std::string& args);
+    void commandDebugMsg_OnOff(int fd, const std::string& args);
+    void commandDirector_pause(int fd, const std::string& args);
+    void commandDirector_resume(int fd, const std::string& args);
+    void commandDirector_stop(int fd, const std::string& args);
+    void commandDirector_start(int fd, const std::string& args);
```
Comment (minggo) [inline cocos/base/CCConsole.h]: It is better to rename commandDirector_start to commandDirectorStart according [cocos2d-x coding style](https://github.com/cocos2d/cocos2d-x/blob/v3/docs/CODING_STYLE.md#function-names). And there are many functions to be modified.

Comment (m-yukio) [inline cocos/base/CCConsole.h]: Would that it contains an underscore in the function name?
I will understand. I'll fix.

Comment (minggo) [inline cocos/base/CCConsole.h]: Yep, it uses `_` to connect words.

**Comment by m-yukio** [comment]
@minggo I was fixed.

**Comment by minggo** [comment]
Thanks @m-yukio , i will check the code locally to view.

**Comment by minggo** [comment]
@m-yukio Thanks. It is a good PR.

**Comment by m-yukio** [comment]
Thank you!

**Pull request merged**